# STAT 207 Lab 7: Hypothesis Testing - [25.5 total points] (0.5) is bonus points - Solutions

Due: Wednesday, March 24 11:59pm CST



## <u>Case Study</u>: U.S Unemployment Rates and Median Education Levels
In this lab we will explore the statistical concepts that we've learned this week while also examining the **unemployment rates** and **median education levels** of the counties in the U.S. 

### Imports

In [1]:
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()                              # sets the default plotting style
import numpy as np

## 1. <u>Preliminaries</u>

### Read the county.csv into a dataframe, call it df, and display the first five rows.

Remember that 'data unavailable' represents missing values in this csv. Make sure you encode these values as NaN when you read the data in.

In [2]:
df = pd.read_csv('county.csv', na_values=['data unavailable'])         #read the file
df.head()

name    state   pop2000  pop2010   pop2017  pop_change  poverty  \
0  Autauga County  Alabama   43671.0    54571   55504.0        1.48     13.7   
1  Baldwin County  Alabama  140415.0   182265  212628.0        9.19     11.8   
2  Barbour County  Alabama   29038.0    27457   25270.0       -6.22     27.2   
3     Bibb County  Alabama   20826.0    22915   22668.0        0.73     15.2   
4   Blount County  Alabama   51024.0    57322   58013.0        0.68     15.6   

   homeownership  multi_unit  unemployment_rate metro    median_edu  \
0           77.5         7.2               3.86   yes  some_college   
1           76.7        22.6               3.99   yes  some_college   
2           68.0        11.1               5.90    no    hs_diploma   
3           82.9         6.6               4.39   yes    hs_diploma   
4           82.0         3.7               4.02   yes    hs_diploma   

   per_capita_income  median_hh_income smoking_ban  
0           27841.70           55317.0        none  
1           27779.85           52562.0        none  
2           17891.73           33368.0     partial  
3           20572.05           43404.0        none  
4           21367.39           47412.0        none

### Create a series that is comprised of just the unemployment rate variable and drop any missing values from this dataset.

In [1]:
u_r = df[['unemployment_rate']].dropna()

NameError: name 'df' is not defined

### Create a series that is comprised of just the median education variable and drop any missing values from this dataset.

In [4]:
m_e = df['median_edu'].dropna()

## 2. <u>Unemployment Rate Population Analysis</u>

We will assume that the series of all county unemployment rates from 1b is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.

### 2a. [0.5 pt] Calculate the mean and standard deviation unemployment rate of ALL U.S. counties.

In [5]:
mean_2a = u_r.mean()
mean_2a

4.611443134756305

In [6]:
std_2a = u_r.std()
std_2a

1.6487610856225392

### 2b.  [1 pt] Calculate the z-score of the unemployment rate of 'Champaign County' in the state of 'Illinois'.

In [7]:
df[df['name']=='Champaign County']

name     state   pop2000  pop2010   pop2017  pop_change  \
604   Champaign County  Illinois  179669.0   201081  209399.0        1.83   
2053  Champaign County      Ohio   38890.0    40097   38840.0       -1.54   

      poverty  homeownership  multi_unit  unemployment_rate metro  \
604      21.4           55.9        35.2               4.24   yes   
2053     11.1           75.3        14.7               4.16    no   

        median_edu  per_capita_income  median_hh_income smoking_ban  
604   some_college           29619.49           49586.0         NaN  
2053    hs_diploma           25704.80           54495.0         NaN

In [8]:
from scipy.stats import norm
p =norm.cdf(4.26, loc=mean_2a, scale = std_2a)
critical_value = norm.ppf(p)
critical_value

-0.21315588887980552

## 3. <u>Unemployment Rate Sample Analysis</u>

### 3a.  [0.5 pt] Collect a random sample of size 35 from your population of all county unemployment rates (without replacement) using a random state of 3.

In [9]:
sample1 = u_r.sample(35,replace = False, random_state = 3)

### 3b.  [0.5 pt] Calculate the mean and standard deviation of this sample.

In [10]:
mean_s = sample1.mean()
print('mean:',mean_s)

mean: 4.252000000000002


In [11]:
std_s = sample1.std()
print('std:',std_s)

std: 1.3095751443974224


## 4. <u>Unemployment Rate Sampling Distribution Analysis</u>

Assume you know $\sigma$ in this part.

### 4a.  [0.5 pt] What is the mean and standard deviation of the sampling distribution of sample mean unemployment rates (collected from samples of size 35)?

it should be the same as the mean of the dataset

In [12]:
mean_4a= mean_2a
mean_4a

4.611443134756305

it should be

In [13]:
std_4a = std_2a/np.sqrt(35)
std_4a

0.2786914893089394

### 4b.  [1 pt] Is the sampling distribution of samples means (collected from samples of size 35) approximately normal? Explain why or why not?

yes, because the each sample size is bigger than 30 and the population distribution is approximately normal

### 4c.  [1 pt] What is the z-score of our sample mean from 3b?

In [14]:
p =norm.cdf(mean_s, loc=mean_4a, scale = std_4a)
critical_value = norm.ppf(p)
critical_value

-1.2897528218303367

### 4d.  [1 pt] What is the probability that a randomly selected sample mean z-score is greater than or equal to the z-score that we calculated in 4c?

In [15]:
1- norm.cdf(mean_s, loc=mean_4a, scale = std_4a)

0.9014317534230483

### 4e.  [0.5 pt] What is the probability that a randomly selected sample mean z-score is less than or equal to the negative absolute value of the z-score that we calculated in 4c?

In [16]:
norm.cdf(mean_s, loc=mean_4a, scale = std_4a)

0.09856824657695179

## 5. <u>Hypothesis Testing for the Average Unemployment Rate of ALL US Counties</u>

For now, suppose that we know nothing about our population that we explored in #2.


Suppose that a reliable source suggested that the average unemployment rate of all US counties is 4.611. We are skeptical of this source, and would like to test this claim and see if there is evidence to suggest that this is not the case. We will use our random sample that we collected from the population in #3 to test this claim.

### 5a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

    null: the average unemployment rate of all US counties is 4.611
    hypothses: the average unemployment rate of all US counties is not 4.611

### 5b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #3.



1. the condition is indepedent: because the sample size(35) is less than 10% of the data size (3142)
2. n =35 > 30

### 5c.  [1 pt] Calculate the p-value for this test.

Assume you do not know $\sigma$.

In [17]:
std_5 = std_s / np.sqrt(35)
std_5

0.2213586010348386

In [18]:
p_value = 2*(norm.cdf(mean_s, loc = 4.611, scale = std_5))
print("p-value:", p_value)

p-value: 0.10484557772754166


### 5d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.05$.

since the p-value = 0.1048 >$\alpha=0.05$ so we fail to reject the null hypotheses. And we say that we  do not have have sufficient evidence to suggest the alternative test(the average unemployment rate of all US counties is not 4.611)
 

### 5e.  [1 pt] Now we would like to make a 95% confidence interval for the average unemployment rate of al U.S. counties. First, check the conditions for creating this confidence interval.

1. the condition is indepedent: because the sample size(35) is less than 10% of the data size (3142)
2. n =35 > 30 and it is chosen from a normal distribution
3. the sample is random chosen

### 5f.  [1 pt] Now create this 95% confidence interval.

Assume you do not know $\sigma$.

In [24]:
cv = norm.ppf(0.975)
upper = mean_s + cv*std_s/np.sqrt(35)
lower = mean_s - cv*std_s/np.sqrt(35)
print(lower,",",upper)

3.8181451143035474 , 4.685854885696456


### 5g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Because the null value 4.611 is inside the 95% confidence interval, so we did not successfully reject the null hypothesis. And we say that we do not have sufficient evidence to suggest the alternative test(the average unemployment rate of all US counties is not 4.611)

### 5h.  [0.5 pt] NOW, using what we *actually* know about the population and population mean, did our hypothesis test *actually* make the right conclusions in 5d and 5g?

Yes, because we know the actural population mean is 4.611 which is our null hypothesis, and in both 5d and 5g we so not have enough evidence to reject the null hypothesis thus they all made the right conclusion

## 6. <u>Median Education Level Population Analysis</u>

We will assume that the series of all county median education levels from 1c is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.



### <u>Tutorial</u>: Conditional Filtering of a Series

In [20]:
import pandas as pd
import numpy as np
small_series=pd.Series(['stat107', 'stat207', 'stat307', 'stat207'])
small_series

0    stat107
1    stat207
2    stat307
3    stat207
dtype: object

We can count the number of entries in a series that meet a certain condition by setting up a condition upon the values in a series and taking the sum of the Boolean values.

In [21]:
small_series=='stat207'

0    False
1     True
2    False
3     True
dtype: bool

In [22]:
np.sum(small_series=='stat207')

2

### 6a.  [0.5 pt] Calculate the proportion of all U.S. counties that have a median education level of 'hs_diploma'.

In [23]:
np.sum(m_e == 'hs_diploma')

1397

## 7. <u>Median Population Sample Analysis</u>

### 7a.  [0.5 pt] Collect a random sample of 40 county median education levels from the population of ALL U.S. county median education levels, (without replacement), using a random state of 100.

In [26]:
sample2 = m_e.sample(40,replace = False, random_state = 100)

### 7b.  [0.5 pt] Calculate the proportion of counties in this sample that have a median education level of high school diploma.

In [27]:
pro = np.sum(sample2 == 'hs_diploma')/sample2.shape[0]
pro

0.5

## 8. <u>Sampling Distribution of Sample Proportions Analysis</u>



### 8a. [1 pt] What is the mean and standard deviation of the sampling distribution of sample propotion of counties with a median education level of high school diploma (collected from samples of size 40)?

Assume you actually know the population proprtion $p$ in part 8.

In [28]:
counts_8 = m_e.value_counts()

In [29]:
mean_8 = counts_8['hs_diploma']/counts_8.sum()
std_8 = np.sqrt(mean_8*(1-mean_8)/40)

In [30]:
print('mean is:', mean_8,'\n','std is:', std_8)

mean is: 0.4449044585987261 
 std is: 0.07857551802528899


### 8b.  [1 pt] What is the z-score of our sample proportion from 7b?

In [31]:
p =norm.cdf(pro, loc=mean_8, scale = std_8)
critical_value = norm.ppf(p)
critical_value

0.7011794867650986

## 9. <u>Hypothesis Testing for the Proportion of ALL U.S. Counties that have a Median Education Level of High School Diploma</u>

For now, suppose that we know nothing about our population that we explored in #6.

Previous studies have suggested that in the past, 45% of U.S. counties have a median education level of a high school diploma. We are skeptical that that is the still the case. We would like to test (using our sample from #7), whether there is evidence to suggest that the the proportion of counties with a median education level of high school diploma is diffrent from 45%.

### 9.a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

    null : mean is 45% 
    hypothses: mean is not 45%

### 9b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #7.



1. the condition is indepedent: because the sample size(40) is less than 10% of the data size
2. n =40 > 30

### 9c.  [1 pt] Calculate the p-value for this test.

Assume you do not know anything about the population proportion (including the value of $p$).

In [36]:
mean_9 = pro
std_9 = np.sqrt(0.45*(1-0.45)/40)

In [37]:
p_value = 2*(1-norm.cdf(mean_9, loc = 0.45, scale = std_9))
print("p-value:", p_value)

p-value: 0.5250099664186034


### 9d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.10$.

since the p-value = 0.525 >$\alpha=0.10$ so we fail to reject the null hypotheses. And we say that we do not have have sufficient evidence to suggest the alternative test(the proportion of counties with a median education level of high school diploma is diffrent from 45%.)
 

### 9e.  [1 pt] Now we would like to make a 90% confidence interval for the proportion of ALL U.S. counties that have a median education level of high school diploma.. First, check the conditions for creating this confidence interval.

1. the sample is random chosen
2. n =40 > 30 and it is chosen from a normal distribution
3. the condition is indepedent: because the sample size(40) is less than 10% of the data size

### 9f.  [2 pt] Now create this 90% confidence interval.

Assume you do not know anything about the population (including $p$).

In [39]:
cv = norm.ppf(0.95)
upper = mean_9 + cv*std_9
lower = mean_9 - cv*std_9
print(lower,",",upper)

0.3706147221581398 , 0.6293852778418603


### 9g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Because the null value 0.45 is inside the 90% confidence interval, so we did not successfully reject the null hypothesis. And we say that we do not have sufficient evidence to suggest the alternative test(the proportion of counties with a median education level of high school diploma is diffrent from 45%)